qutebrowser/components/misccommands.py

```diff
@@ -223,41 +223,80 @@ def _insert_text_cb(elem: Optional[apitypes.WebElement]) -> None:
-@cmdutils.argument('filter_', choices=['id'])
-def click_element(tab: apitypes.Tab, filter_: str, value: str, *,
+@cmdutils.argument('filter_', choices=['id', 'css', 'position', 'focused'])
+def click_element(tab: apitypes.Tab, filter_: str, value: str = None, *,
                   target: apitypes.ClickTarget =
                   apitypes.ClickTarget.normal,
-                  force_event: bool = False) -> None:
+                  force_event: bool = False,
+                  select_first: bool = False) -> None:
     """Click the element matching the given filter.
 
     The given filter needs to result in exactly one element, otherwise, an
     error is shown.
 
     Args:
         filter_: How to filter the elements.
-                 id: Get an element based on its ID.
-        value: The value to filter for.
+
+            - id: Get an element based on its ID.
+            - css: Filter by a CSS selector.
+            - position: Click the element at specified position.
+                Write value as 'x,y'.
+            - focused: Click the currently focused element.
+        value: The value to filter for. Optional for 'focused' filter.
         target: How to open the clicked element (normal/tab/tab-bg/window).
         force_event: Force generating a fake click event.
+        select_first: Select first matching element if there are multiple.
     """
-    def single_cb(elem: Optional[apitypes.WebElement]) -> None:
-        """Click a single element."""
-        if elem is None:
-            message.error("No element found with id {}!".format(value))
-            return
+    def do_click(elem: apitypes.WebElement) -> None:
         try:
             elem.click(target, force_event=force_event)
         except apitypes.WebElemError as e:
             message.error(str(e))
+
+    def single_cb(elem: Optional[apitypes.WebElement]) -> None:
+        """Click a single element."""
+        if elem is None:
+            message.error(f"No element found matching {filter_}={value}!")
+            return
+
+        do_click(elem)
+
+    def multiple_cb(elems: List[apitypes.WebElement]) -> None:
+        if not elems:
+            message.error(f"No element found matching {filter_}={value}!")
+            return
+
+        if not select_first and len(elems) > 1:
+            message.error(f"Multiple elements found matching {filter_}={value}!")
             return
 
+        do_click(elems[0])
+
+    if value is None and filter_ != 'focused':
+        raise cmdutils.CommandError("Argument 'value' is only"
+                                    "optional with filter 'focused'!")
+
     handlers = {
-        'id': (tab.elements.find_id, single_cb),
+        'id': (tab.elements.find_id, value, single_cb),
+        'css': (tab.elements.find_css, value, multiple_cb, message.error),
+        'position': (_wrap_find_at_pos, value, tab, single_cb),
+        'focused': (tab.elements.find_focused, single_cb),
```

Comment:
Maybe switch this dict to storing lambdas instead, so that keyword arguments can be used where appropriate? Something like: `lambda: tab.elements.find_css(selector=value, callback=single_cb, error_cb=message.error)`.

Or go the KISS route and just convert this to an `if`, I suppose.